### DISCUSSION TITLE
ipsec.secrets file included in strongswan.conf not read

### DISCUSSION THREAD

**Discussion by jens-uyttersprot** [strongswan/strongswan#611] [Q&A]
Hello,

I have configured the /etc/strongswan.conf file to use /strongswan/ipsec.conf as the configuration file and the /strongswan/ipsec.secrets file as the file which contains all the secrets. The ipsec.conf file is read correctly but the secrets are not.

/etc/strongswan.conf
```
# strongswan.conf - strongSwan configuration file
#
# Refer to the strongswan.conf(5) manpage for details
#
# Configuration changes should be made in the included files

starter {
    config_file = /strongswan/ipsec.conf
}

charon {
	load_modular = yes
	plugins {
        stroke {
            secrets_file = /strongswan/ipsec.secrets
        }
		include strongswan.d/charon/*.conf
	}
}

include strongswan.d/*.conf
```

If I try to initiate a tunnel configured in the /strongswan/ipsec.conf file with `ipsec up tunnel1` the error message shows that there are no secrets for that connection. When I try to move the secrets file to /etc/ipsec.secrets and initiate the connection, it works perfectly. 

I also tried to place the `secrets_file = /strongswan/ipsec.secrets` line in the /etc/strongswan.d/charon/stroke.conf file but that also doesn't work. After every config change I do an ipsec restart to load the newly configured files.

Is there anyone who could help me with this issue as I have already troubleshooted this problem for a long time.

Thanks in advance!

Kind regards,
Jens Uyttersprot

**Comment by tobiasbrunner** [comment]
Read the log (see #196).
